### PR TITLE
Changes related to adding mhd analytic solutions in Diskpop

### PR DESCRIPTION
**Pull request by alicesomigliana**
Modifications needed to add MHD analytic solutions in Diskpop.

### CODE REVIEW

File: src/ViscousEvolution.py [hunk 1]
```diff
@@ -166,6 +166,35 @@ def __call__(self, R, t):
         
         return self._Sigma0 * ft * Xg * np.exp( - Xg*X*X / tt)
         
+#########################
+
+class PL_sigma(object):
+    """
+    Simple power-law solution. Dased on Lynden-Bell & Pringle (1974), without the exponential cutoff.
```
Comment (giovanni-rosotti) [inline src/ViscousEvolution.py]: Typo dased->based

File: src/disc.py [hunk 2]
```diff
@@ -123,6 +123,17 @@ def dust_frac(self):
     def amax(self):
         return 0
 
+    def new_Omega_k(self, R):
```
Comment (giovanni-rosotti) [inline src/disc.py]: Do we still need this? I thought this was something used in debugging but that we no longer need?

File: src/disc.py [hunk 3]
```diff
@@ -123,6 +123,17 @@ def dust_frac(self):
...
+        return np.sqrt(self._star.M/R**3)
+
+    def mass(self):
```
Comment (giovanni-rosotti) [inline src/disc.py]: If we already know we are going to rename this function to disc_mass we may just as well do it now

File: src/disc.py [hunk 4]
```diff
@@ -123,6 +123,17 @@ def dust_frac(self):
...
+        """
+        Compute the disc mass in Msun.
+        """
+        mass_tmp = 2. * np.pi*np.trapz(self.R*self.Sigma, self.R)
+        mass = mass_tmp*(1.5e13)**2/2e33
```
Comment (giovanni-rosotti) [inline src/disc.py:131]: Can you please add a short comment (just 1-2 lines) on where this magic factor is coming from? (mea culpa I should have done it also in the other places where this is used...) Or if there's something in the documentation where we say it just say "see documentation"

File: src/ViscousEvolution.py [hunk 5]
```diff
@@ -166,6 +166,35 @@ def __call__(self, R, t):
...
         
+#########################
+
+class PL_sigma(object):
+    """
+    Simple power-law solution. Dased on Lynden-Bell & Pringle (1974), without the exponential cutoff.
+    """
+    
+    def __init__(self, M, rc, nuc, gamma=1):
+        self._rc  = rc
```
Comment (giovanni-rosotti) [inline src/ViscousEvolution.py:177]: What's the meaning of M now? It used to be the disc mass, but without truncation the formula is no longer correct. Also rc does not make much sense anymore. Also in _call_ it doesn't make much sense to have the tt factor.

**Review by giovanni-rosotti — COMMENTED** [review]
Most things are really small and you can do in 5 minutes. I think we need some thinking instead about the interface of the power-law solution. Is the idea to use it to evolve things in time? In this case the time evolution is not correct with the tt factor, because that comes from the exponentially truncated solution, and also the mass normalisation is not correct. If it's needed only for the initial conditions instead, then it doesn't need to know anything at all about time...

File: src/ViscousEvolution.py (hunk 1 above)
Comment (alicesomigliana) [reply src/ViscousEvolution.py]: Fixed

File: src/disc.py (hunk 4 above)
Comment (alicesomigliana) [reply src/disc.py:131]: Added

File: src/disc.py (hunk 3 above)
Comment (alicesomigliana) [reply src/disc.py]: Done

File: src/disc.py (hunk 2 above)
Comment (alicesomigliana) [reply src/disc.py]: True, we don't - removed

File: src/ViscousEvolution.py (hunk 5 above)
Comment (alicesomigliana) [reply src/ViscousEvolution.py:177]: Resolved as discussed on Slack

**Pull request merged**